Sources/StreamChatUI/ChatMessageList/ChatMessageListView.swift

```diff
@@ -175,6 +175,54 @@ open class _ChatMessageListView<ExtraData: ExtraDataTypes>: UITableView, Customi
         completion: (() -> Void)? = nil
     ) {
         var shouldScrollToBottom = false
+        
+        // The verification of conflicts is just a temporary solution
+        // until the root cause of the conflicts has been solved.
+        var allIndexes = Set<IndexPath>()
+        var moveIndexes = Set<IndexPathMove>()
+        var insertIndexes = Set<IndexPath>()
+        var removeIndexes = Set<IndexPath>()
+        var updateIndexes = Set<IndexPath>()
+        
+        var hasConflicts = false
+        let verifyConflict = { (indexPath: IndexPath) in
+            let (inserted, _) = allIndexes.insert(indexPath)
+            hasConflicts = !inserted || hasConflicts
+        }
+        
+        for change in changes {
+            if hasConflicts {
+                break
+            }
+            
+            switch change {
+            case let .insert(_, index):
+                verifyConflict(index)
+                insertIndexes.insert(index)
+            case let .move(_, fromIndex, toIndex):
+                verifyConflict(fromIndex)
+                verifyConflict(toIndex)
+                moveIndexes.insert(IndexPathMove(fromIndex, toIndex))
+            case let .remove(_, index):
+                verifyConflict(index)
+                removeIndexes.insert(index)
+            case let .update(_, index):
+                verifyConflict(index)
+                updateIndexes.insert(index)
+            }
+        }
+        
+        if hasConflicts {
+            reloadData()
+            
+            logConflicts(
+                moves: moveIndexes,
+                inserts: insertIndexes,
+                updates: updateIndexes,
+                removes: removeIndexes
+            )
+            return
+        }
```

Comment:
Maybe it is better to extract this to an internal component and use it in both places? 🤔